core/src/main/scala/com/eed3si9n/jarjarabrams/scalasig/ScalaSigAnnotationVisitor.scala

```diff
@@ -127,3 +111,22 @@ class ScalaSigAnnotationVisitor(
     ca
   }
 }
+
+object ScalaSigAnnotationVisitor {
+  def nextChunk(chars: Array[Char], from: Int, maxBytes: Int): Array[Char] = {
+    if (from == chars.length) {
+      Array.empty
+    } else {
+      var size = 0
+      var index = 0
+
+      while (size < maxBytes && from + index < chars.length) {
+        val c = chars(from + index)
+        size += (if (c == 0) 2 else 1)
+        if (size <= maxBytes) index += 1
```

Comment:
This a bug fix. Originally it was just `index += 1` without condition.  So it was able to produce chunk that was MaxStringSizeInBytes + 1 when encounter zero byte at end of chunk.

Reply:
That makes sense. Thanks for the explanation.